internal/tools/docthebuilder/cmd/validate.go

```diff
@@ -0,0 +1,87 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tools/docthebuilder/data"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tools/docthebuilder/rule"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tools/docthebuilder/util"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tools/docthebuilder/validator"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+var (
+	vRulesFlag    string
+	vServiceFlag  string
+	vResourceFlag string
+
+	validateCmd = &cobra.Command{
+		Use:   "validate",
+		Short: "Validates resource documentation",
+		Run: func(cmd *cobra.Command, args []string) {
+			var err error
+
+			util.InitLogger(debugFlag)
+			fs := afero.NewOsFs()
+
+			if providerDirFlag == "" {
+				providerDirFlag, err = os.Getwd()
+				if err != nil {
+					log.WithError(err).Fatal("retrieving current working directory")
+				}
+			}
+
+			if !util.DirExists(fs, providerDirFlag) {
+				log.WithField("path", providerDirFlag).Fatal("unable to access provider directory")
+			}
+
+			rules := util.MapKeys2Slice(rule.Registration)
+			if vRulesFlag != "" {
+				rules = strings.Split(vRulesFlag, ",")
+			}
```

Comment:
also if the flags are then moved to their own struct that is initialized you could move this code there? ie https://github.com/katbyte/ghp-repo-sync/blob/898754f63eb151966a586b8b93d606530ab7b4cd/cli/flags.go#L131